### ISSUE TITLE
E2E Cluster creation is unstable

### ISSUE THREAD

**Issue by barkbay** [elastic/cloud-on-k8s#2461]
https://devops-ci.elastic.co/blue/organizations/jenkins/cloud-on-k8s-stack/detail/cloud-on-k8s-stack/162/pipeline/36

![image](https://user-images.githubusercontent.com/976373/72966548-7fc6bc00-3dbf-11ea-834f-ca88bfad672d.png)

Only the creation of the `6.8.x` cluster has succeeded. All others have failed with the following errors:

* 7.1.1
```
[2020-01-23T00:50:00.464Z] ERROR: (gcloud.beta.container.clusters.create) Operation [<Operation
[2020-01-23T00:50:00.465Z]  clusterConditions: []
[2020-01-23T00:50:00.465Z]  detail: u'Retry budget exhausted (80 attempts): The network "default" does not have available private IP space in 10.0.0.0/8 to reserve a /14 block for pods for cluster {europe-west2, 1014491842772, ****, eck-71-162-e2e, 83cb6be5dc25bfe91a9e468b8975ebc9dbc63b3c6635d033d3aca7156ea6570a}.'
[2020-01-23T00:50:00.465Z]  endTime: u'2020-01-23T00:49:54.093175723Z'
[2020-01-23T00:50:00.465Z]  name: u'operation-1579738810624-f03fae70'
[2020-01-23T00:50:00.465Z]  nodepoolConditions: []
[2020-01-23T00:50:00.465Z]  operationType: OperationTypeValueValuesEnum(CREATE_CLUSTER, 1)
[2020-01-23T00:50:00.465Z]  progress: <OperationProgress
[2020-01-23T00:50:00.465Z]  metrics: [<Metric
[2020-01-23T00:50:00.465Z]  intValue: 12
[2020-01-23T00:50:00.465Z]  name: u'CLUSTER_CONFIGURING'>, <Metric
[2020-01-23T00:50:00.465Z]  intValue: 12
[2020-01-23T00:50:00.465Z]  name: u'CLUSTER_CONFIGURING_TOTAL'>]
[2020-01-23T00:50:00.465Z]  stages: []>
[2020-01-23T00:50:00.465Z]  selfLink: u'https://container.googleapis.com/v1beta1/projects/1014491842772/locations/europe-west2/operations/operation-1579738810624-f03fae70'
[2020-01-23T00:50:00.465Z]  startTime: u'2020-01-23T00:20:10.624807528Z'
[2020-01-23T00:50:00.465Z]  status: StatusValueValuesEnum(DONE, 3)
[2020-01-23T00:50:00.465Z]  statusMessage: u'Retry budget exhausted (80 attempts): The network "default" does not have available private IP space in 10.0.0.0/8 to reserve a /14 block for pods for cluster {europe-west2, 1014491842772, ****, eck-71-162-e2e, 83cb6be5dc25bfe91a9e468b8975ebc9dbc63b3c6635d033d3aca7156ea6570a}.'
[2020-01-23T00:50:00.465Z]  targetLink: u'https://container.googleapis.com/v1beta1/projects/1014491842772/locations/europe-west2/clusters/eck-71-162-e2e'
[2020-01-23T00:50:00.465Z]  zone: u'europe-west2'>] finished with error: Retry budget exhausted (80 attempts): The network "default" does not have available private IP space in 10.0.0.0/8 to reserve a /14 block for pods for cluster {europe-west2, 1014491842772, ****, eck-71-162-e2e, 83cb6be5dc25bfe91a9e468b8975ebc9dbc63b3c6635d033d3aca7156ea6570a}.
[2020-01-23T00:50:00.465Z] Error: exit status 1
```
* For the others:

```
[2020-01-23T00:50:32.555Z] Timed out waiting for operation <Operation
[2020-01-23T00:50:32.555Z]  clusterConditions: []
[2020-01-23T00:50:32.555Z]  detail: u'Cluster is being configured'
[2020-01-23T00:50:32.555Z]  name: u'operation-1579738818187-872ca988'
[2020-01-23T00:50:32.555Z]  nodepoolConditions: []
[2020-01-23T00:50:32.555Z]  operationType: OperationTypeValueValuesEnum(CREATE_CLUSTER, 1)
[2020-01-23T00:50:32.555Z]  progress: <OperationProgress
[2020-01-23T00:50:32.555Z]  metrics: [<Metric
[2020-01-23T00:50:32.555Z]  intValue: 11
[2020-01-23T00:50:32.555Z]  name: u'CLUSTER_CONFIGURING'>, <Metric
[2020-01-23T00:50:32.555Z]  intValue: 12
[2020-01-23T00:50:32.555Z]  name: u'CLUSTER_CONFIGURING_TOTAL'>]
[2020-01-23T00:50:32.555Z]  stages: []>
[2020-01-23T00:50:32.555Z]  selfLink: u'https://container.googleapis.com/v1beta1/projects/1014491842772/locations/europe-west2/operations/operation-1579738818187-872ca988'
[2020-01-23T00:50:32.555Z]  startTime: u'2020-01-23T00:20:18.187055534Z'
[2020-01-23T00:50:32.555Z]  status: StatusValueValuesEnum(RUNNING, 2)
[2020-01-23T00:50:32.555Z]  targetLink: u'https://container.googleapis.com/v1beta1/projects/1014491842772/locations/europe-west2/clusters/eck-72-162-e2e'
[2020-01-23T00:50:32.555Z]  zone: u'europe-west2'>
[2020-01-23T00:50:32.555Z] ERROR: (gcloud.beta.container.clusters.create) Operation [<Operation
[2020-01-23T00:50:32.555Z]  clusterConditions: []
[2020-01-23T00:50:32.555Z]  detail: u'Cluster is being configured'
[2020-01-23T00:50:32.555Z]  name: u'operation-1579738818187-872ca988'
[2020-01-23T00:50:32.555Z]  nodepoolConditions: []
[2020-01-23T00:50:32.555Z]  operationType: OperationTypeValueValuesEnum(CREATE_CLUSTER, 1)
[2020-01-23T00:50:32.555Z]  progress: <OperationProgress
[2020-01-23T00:50:32.555Z]  metrics: [<Metric
[2020-01-23T00:50:32.555Z]  intValue: 11
[2020-01-23T00:50:32.555Z]  name: u'CLUSTER_CONFIGURING'>, <Metric
[2020-01-23T00:50:32.555Z]  intValue: 12
[2020-01-23T00:50:32.555Z]  name: u'CLUSTER_CONFIGURING_TOTAL'>]
[2020-01-23T00:50:32.555Z]  stages: []>
[2020-01-23T00:50:32.555Z]  selfLink: u'https://container.googleapis.com/v1beta1/projects/1014491842772/locations/europe-west2/operations/operation-1579738818187-872ca988'
[2020-01-23T00:50:32.555Z]  startTime: u'2020-01-23T00:20:18.187055534Z'
[2020-01-23T00:50:32.555Z]  status: StatusValueValuesEnum(RUNNING, 2)
[2020-01-23T00:50:32.555Z]  targetLink: u'https://container.googleapis.com/v1beta1/projects/1014491842772/locations/europe-west2/clusters/eck-72-162-e2e'
[2020-01-23T00:50:32.555Z]  zone: u'europe-west2'>] is still running
[2020-01-23T00:50:32.555Z] Error: exit status 1
[2020-01-23T00:50:32.555Z] exit status 1
```

Also the clusters are not deleted:

![image](https://user-images.githubusercontent.com/976373/72966888-404c9f80-3dc0-11ea-9738-a715cb0c9d39.png)

**Comment by sebgl** [comment]
I think we can either:

1. create and delete one VPC network per cluster (I don't think this would impact GCP pricing), since we want them isolated anyway
2. use a dedicated VPC for Jenkins clusters

1. looks like a better long-term solution.

**Comment by sebgl** [comment]
I think we can also restrict the [CIDR block range of the Pods and Services secondary IP address ranges](https://cloud.google.com/kubernetes-engine/docs/how-to/alias-ips?hl=en). Which may be the simpler option to start with. The default `/14` range for Pods and `/20` for services seems overkill for our use case.

**Comment by sebgl** [comment]
I think https://github.com/elastic/cloud-on-k8s/pull/2473 should give us more breathing room. We may want to recreate existing clusters so they apply the same CIDR range.
We can re-evaluate creating a VPC network per cluster later if this turns out to not be enough.

**Comment by barkbay** [comment]
Seems to be fixed by #2473 , thanks @sebgl !
